File: Data Structures & Algorithms/car-fleet/submission-0.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        int m = position.size();
        vector<pair<int, int>> x;
        stack<double> ans;
        for(int i{0}; i < position.size(); ++i){
            x.push_back({position[i], speed[i]});
        }
        sort(x.begin(), x.end());
        for(int i = m-1; i >= 0; i--){
            double time = (double)(target - x[i].first) / x[i].second;
            if(ans.empty() || time > ans.top()){
                ans.push(time);
            }
        }
        return ans.size();
    }
};

File: Data Structures & Algorithms/reverse-bits/submission-0.java
class Solution {
    public int reverseBits(int n) {
        int ans = 0;
        int count = 0;
        while(count < 32){
            int bit = n & 1;
            n = n >>> 1;
            ans = ans << 1;
            ans |= bit;
            count++;
        }
        return ans;
    }
}
